Week 3: 권지윤 수업내용, 과제 제출

jiyoxn opened the pull request:
Feat: 과제1
-SignUp 페이지 구현
-ConstraintLayout에 적합한 방식으로 구현
-Layout 단위에 Margin or Padding 적용
Feat: 과제2
-계산기 구현
-숫자 두개를 입력
-사칙연산 기호를 누르면 그 결과가 출력
-숫자만 있는 키패드가 생성
-/ 의 경우 두번째 숫자가 0이면
-"Result : -"를 출력
-Toast로 "0으로 나눌 수 없습니다." 출력
Feat: study
-Week3 수업 실습

jiyoxn committed:
Week 3: 권지윤 수업내용, 과제 제출

Feat: 과제1
-SignUp 페이지 구현
-ConstraintLayout에 적합한 방식으로 구현
-Layout 단위에 Margin or Padding 적용
Feat: 과제2
-계산기 구현
-숫자 두개를 입력
-사칙연산 기호를 누르면 그 결과가 출력
-숫자만 있는 키패드가 생성
-/ 의 경우 두번째 숫자가 0이면
-"Result : -"를 출력
-Toast로 "0으로 나눌 수 없습니다." 출력
Feat: study
-Week3 수업 실습

chattymin commented on 권지윤/Week 3/hw1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,140 @@
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_margin="20dp"
+    android:paddingBottom="30dp"
```
마진과 패딩의 차이점을 알고있나요??
혼용해서 사용하신 이유가 궁금하네요

jiyoxn replied:
android:layout_margin="20dp" 한 상태에서 layout_marginBottom으로 아래에서만 더 띄우려다가 마진으로 띄우려면 start, end, bottom 각각 줘야하는것같아서 패딩을 주었습니다.. 각각 주는걸로 바꿨어요!!

chattymin commented on 권지윤/Week 3/hw1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,140 @@
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:ignore="MissingConstraints" />
```
이 코드는 위치를 지정해주지 않았을때 생기는 오류를 무시해주는 코드입니다.
현재 코드에서는 위치를 전부 설정해줬기 때문에 없어도 되는 코드 같네요

jiyoxn replied:
다 없애면 위치가 중앙에서 벗어나길래 tools:ignore="MissingConstraints"만 지웠어요!

chattymin commented on 권지윤/Week 3/hw1/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,140 @@
+            android:layout_marginBottom="12dp"
+            android:ems="10"
+            android:inputType="text"
+            android:text="아이디를 입력하세요"
```
text에 해당 값이 있을 경우, EditText를 클릭했을 때 위 값이 남아있게 됩니다.
"hint"의 사용법을 확인 후 수정하시면 좋을 것 같네요 :)

jiyoxn replied:
넵 감사합니다 :)

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/java/com/jiyoxn/calhw/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        input1 = findViewById(R.id.num1)
+        input2 = findViewById(R.id.num2)
+        resultText = findViewById(R.id.result)
```
꾸짖을 갈!!!!!!!!!!!

binding으로 바꿔오세요 ^^

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/java/com/jiyoxn/calhw/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+        val num1Str = input1.text.toString()
+        val num2Str = input2.text.toString()
+
+            val num1 = num1Str.toInt()
+            val num2 = num2Str.toInt()
+            var result = 0
+
+            when (operation) {
+                "add" -> result = num1 + num2
+                "sub" -> result = num1 - num2
+                "mul" -> result = num1 * num2
+                "div" -> {
+                    if (num2 != 0) {
+                        result = num1 / num2
+                    } else {
+                        showToast("0으로 나눌 수 없습니다")
+                        resultText.text="result: - "
+                        return
+                    }
+                }
+            }
+
+            resultText.text = "result: $result"
```
tab이 한칸 더들어가 있네요
이런 경우는 "cmd + option + L"을 누르시면 자동으로 맞춰준답니다. 애용하셔용

jiyoxn replied:
맥린이 맞춤 코멘트 감사합니다

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/java/com/jiyoxn/calhw/MainActivity.kt:
```diff
@@ -0,0 +1,71 @@
+                "sub" -> result = num1 - num2
+                "mul" -> result = num1 * num2
+                "div" -> {
+                    if (num2 != 0) {
+                        result = num1 / num2
+                    } else {
+                        showToast("0으로 나눌 수 없습니다")
+                        resultText.text="result: - "
+                        return
+                    }
```
이 부분은 약간 수정하면 더 좋을거 같아요. 아래와 같은 방식으로 하면 코드가 조금 더 가독성이 좋을 것같네요
조건문(if)내부에는 가능한 true가 오게 짜는게 더 좋아용


if (num2 == 0) {
    showToast("0으로 나눌 수 없습니다")
    resultText.text="result: - "
    return
} 

result = num1 / num2

jiyoxn replied:
헉쓰 감사합니다

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,87 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:ignore="MissingConstraints" />
```
여기두 이거 없어도 돼용~

jiyoxn replied:
여기도 tools:ignore="MissingConstraints" 이거만 지웠어용

chattymin reviewed (changes requested):
열시미 하셨네요!! 고생하셨습니당~~

jiyoxn committed:
Week 3: 권지윤 과제 수정 제출

Refactor: 코드 수정
-마진, 패딩 혼용
-필요없는 코드 지우기
-hint사용
-Binding 사용하기
-"cmd + option + L"
-조건문 안에 not 수정

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/java/com/jiyoxn/calhw/MainActivity.kt:
```diff
@@ -0,0 +1,54 @@
+        val view = binding.root
+        setContentView(view)
+
+        binding.num1.inputType = android.text.InputType.TYPE_CLASS_NUMBER
+        binding.num2.inputType = android.text.InputType.TYPE_CLASS_NUMBER
```
요건 무슨 기능인가요...?
왜 넣은거징(이사람 진짜 모름)

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/java/com/jiyoxn/calhw/MainActivity.kt:
```diff
@@ -0,0 +1,54 @@
+        binding.div.setOnClickListener { performOperation("div") }
+    }
+
+    private fun performOperation(operation: String) {
+        val input1 = binding.num1.text.toString()
+        val input2 = binding.num2.text.toString()
+
+        val num1 = input1.toInt()
+        val num2 = input2.toInt()
+        var result = 0
+
+        when (operation) {
+            "add" -> result = num1 + num2
+            "sub" -> result = num1 - num2
+            "mul" -> result = num1 * num2
+            "div" -> {
+                if (num2 == 0) {
+                    showToast("0으로 나눌 수 없습니다")
+                    binding.result.text = "result: - "
+                    return
+                }
+                result = num1 / num2
+            }
+        }
+        binding.result.text = "result: $result"
+    }
```
함수활용 폼 미쳤다

chattymin commented on 권지윤/Week 3/hw2/CalHw/app/src/main/res/layout/activity_main.xml:
```diff
@@ -0,0 +1,84 @@
+        android:layout_marginTop="24dp"
+        android:ems="10"
+        android:inputType="text"
+        android:text=""
```
빈 값이면 굳이 설정 안해도 된답니다

chattymin reviewed (approved):
정말 열심히 과제한게 눈에 보이네요!!
고생하셨습니당!
